pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -32,6 +32,19 @@ pub mod pallet {
 	use frame_support::pallet_prelude::*;
 	use frame_system::{ensure_root, ensure_signed, pallet_prelude::OriginFor};
 
+	#[derive(Debug, Eq, PartialEq, Encode, Decode, Clone, MaxEncodedLen, TypeInfo)]
+	pub struct SeasonMetadata {
+		pub name: [u8; 100],
+		pub description: [u8; 1000],
```

Comment:
Using arrays might cause panic. It's better to use `Vec<u8>` and check length inside the extrinsic:
```rust
ensure!(season_metadata.name.len() <= 100, Error::<T>::SeasonNameTooLong);
```

Reply:
It's not possible to use `Vec` inside the struct, since it does not implement `MaxEncodedLen`, instead I've used `BoundedVec` with a const bound, since using the bound through the Config also had its own share of problems.

Reply:
Thanks @DidacSF that's it!